ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/ReplLoadTask.java

```diff
@@ -166,14 +185,24 @@ a database ( directory )
           }
           loadTaskTracker.update(functionsTracker);
           functionsTracker.debugLog("functions");
+
+          // Add ReplStateLogTask as function load tasks are always fully added in current cycle
+          loadTaskTracker.updateTaskCount(createFunctionReplLogTask(functionsTracker.tasks(),
+                  iterator.replLogger(), loadFunction.functionName()));
```

Comment:
I don't agree that having it in ReplLoadtask is unreadable/complicated. Will try to move it inside LoadTable.tasks(), LoadPartitions.tasks() and LoadFunction.tasks() but I think, this is complicated as we have different code flow for newTable and existingTable which duplicates the code in several places which could be complicated. Also, it is not feasible to move it inside LoadDatabase, as the end log should be added after adding all tables and functions tasks. So, endLog will be added in ReplLoadtask itself.